pkg/recipes/types.go

```diff
@@ -161,39 +160,20 @@ func GetSecretStoreID(envConfig Configuration, templatePath string) (string, err
 func GetGitURL(templatePath string) (*url.URL, error) {
 	paths := strings.Split(templatePath, "git::")
 	gitUrl := paths[len(paths)-1]
+	if !hasURLScheme(gitUrl) {
+		gitUrl = fmt.Sprintf("https://%s", gitUrl)
+	}
+
 	url, err := url.Parse(gitUrl)
+
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse git url %s : %w", gitUrl, err)
 	}
 
 	return url, nil
 }
 
-// GetEnvAppResourceNames returns the application, environment and resource names.
-func GetEnvAppResourceNames(resourceMetadata *ResourceMetadata) (string, string, string, error) {
-	app, err := resources.ParseResource(resourceMetadata.ApplicationID)
-	if err != nil {
-		return "", "", "", err
-	}
-
-	env, err := resources.ParseResource(resourceMetadata.EnvironmentID)
-	if err != nil {
-		return "", "", "", err
-	}
+func hasURLScheme(url string) bool {
```

Comment:
Have you added unit tests for this function?

Reply:
added tests